doc/libpmem2/pmem2_async.3.md

```diff
@@ -43,30 +45,33 @@ struct vdm_operation_future pmem2_memset_async(struct pmem2_map *map,
 To use those functions, you must have *libminiasync* installed. Those functions use futures
 and vdm (virtual data mover) concepts from this library. Please check **miniasync**(7) for more details.
 
+The struct **pmem2_future** is a structure describing a task to be done asynchronously taking into account persistence
+of the operation. It means that by the time the future is complete, all the data is safely written into a persistent domain.
+
 The **pmem2_config_set_vdm** sets a vdm structure in the *pmem2_config*.
-This structure will be used by pmem2_*_async functions, to create a *vdm_operation_future*.
+This structure will be used by pmem2_*_async functions, to create a *pmem2_future*.
 If vdm is not set in the config, pmem2_map_new will use a default one which uses a
-pmem2 memory movement functions to perform memory operations. (**pmem2_get_memcpy_fn**(3), **pmem2_get_memmove_fn**(3), **pmem2_get_memsety_fn**(3)).
+pmem2 memory movement functions to perform memory operations. (**pmem2_get_memcpy_fn**(3), **pmem2_get_memmove_fn**(3), **pmem2_get_memset_fn**(3)).
 
-The **pmem2_memcpy_async** uses *vdm* structure held inside the *pmem2_map* structure to initialise and returns **vdm_operation_future**.
+The **pmem2_memcpy_async** uses *vdm* structure held inside the *pmem2_map* structure to initialise and returns **pmem2_future**.
 This future will perform memcpy operation defined in *vdm* to copy *len* bytes from *src* to *pmemdest*. In the current implementation *flags* are ignored.
```

Comment:
Since, from what I understand, the `flags` are no longer ignored, this part of the documentation could be modified.